packages/eventual-send/src/E.js

```diff
@@ -304,24 +304,69 @@ export default makeE;
 
 /**
  * The awaited return type of a function.
+ * For the eventual result of an E call, @see {EResult} or @see {ECallableReturn}
  *
  * @template {(...args: any[]) => any} T
  * @typedef {T extends (...args: any[]) => infer R ? Awaited<R> : never} EReturn
  */
 
 /**
- * @template {import('./types.js').Callable} T
+ * An eventual value where remotable objects are recursively mapped to Remote types
+ *
+ * @template T
+ * @typedef {Awaited<T>} EResult
+ */
+
+/**
+ * Experimental type mapping remotable objects to Remote types
+ *
+ * @template T
  * @typedef {(
- *   ReturnType<T> extends PromiseLike<infer U>                       // if function returns a promise
- *     ? T                                                            // return the function
- *     : (...args: Parameters<T>) => Promise<EReturn<T>>  // make it return a promise
+ *   0 extends (1 & T)
+ *     ? T
+ *     : T extends RemotableBrand<infer L, infer P>
+ *       ? (P | RemotableBrand<L, P>)
+ *       : T extends PromiseLike<infer U>
+ *         ? Promise<EAwaitedResult<Awaited<T>>>
+ *         : T extends (null | undefined | string | number | boolean | symbol | bigint | Callable) // Intersections of these types with objects are not mapped
```

Comment:
Is a so-called `FarFunction` a `Callable` and a remotable? Is this branch consistent with that?

Reply:
A far function would be a remotable, and thus match the `RemotableBrand` case earlier. This is the case where we're supporting eventual-send with results containing a pure local (non remotable) functions, and simply passing them through at the type level. Added a comment to clarify